### ISSUE TITLE
Support nightly releases

### ISSUE THREAD

**Issue by thymikee** [react-native-community/upgrade-helper#325]
# Feature Request

`react-native` package outputs `nightly` tag for a while (https://www.npmjs.com/package/react-native), which, if I understand correctly, was recently fixed to work more reliably: https://github.com/facebook/react-native/pull/34838.

Would it be possible for Upgrade Helper to support diffing nightly releases? I would imagine having a separate select to not obscure regular users who are rather not interested in nightlies.

## Why it is needed

To ease working on edgy stuff from RN main for developers.

## Possible implementation

-

### Code sample

-

**Comment by cortinico** [comment]
> Would it be possible for Upgrade Helper to support diffing nightly releases? I would imagine having a separate select to not obscure regular users who are rather not interested in nightlies.

Yeah it should be possible. We would have to invoke the differ job as we invoke the nightly every day.

**Comment by cortinico** [comment]
Nope we're not working on this internally @fabioh8010 👍

**Comment by fabioh8010** [comment]
Great! I will start working on this next week probably, thanks!

**Comment by kelset** [comment]
Hey folks — I’m doing a maintenance pass on the repo and catching up on older issues.

At the time this was opened, nightly diff support felt like a reasonable thing to explore. Since then, though, nightlies have settled into a somewhat different role in the React Native ecosystem.

Today, nightlies are better supported through dedicated nightly integration testing and visibility tooling, rather than through template-to-template upgrade diffs. We now have a dedicated `nightly-tests` repo and website for that work:

- https://github.com/react-native-community/nightly-tests
- https://react-native-community.github.io/nightly-tests/

On top of that, the release cadence is faster than it used to be, and Upgrade Helper already supports the normal stable flow plus RCs. In practice, I think that’s the better place for diff-driven upgrade guidance.

Adding nightly support here would also create a lot of churn and maintenance overhead across both `upgrade-helper` and `rn-diff-purge`:
- daily or near-daily diff generation
- release-list churn in the UI
- cleanup of ephemeral nightly versions
- extra UX complexity for a feature that now looks fairly niche

Given all that, I’m going to close this one.

For pre-release diffing, I’d rather encourage use of the normal stable releases and RCs here, and use the nightly testing infrastructure for nightly-specific signal. If the ecosystem shifts again and there’s a strong concrete need for nightly template diffs, we can always reopen the conversation with updated use cases.
